python/tvm/relay/frontend/onnx.py

```diff
@@ -249,23 +249,39 @@ def flatten_to_nd(x, x_shape, nd=3):
             return out
 
         # Determine the output batch dimension.
+        new_a_shape = a_shape
```

Comment:
I would prefer a less imperative style here; e.g. make it so each if/else branch assigns `new_a_shape` and `new_b_shape` so the branches aren't mutating state up here